update_image.pl

```diff
@@ -359,6 +370,34 @@ sub processConvergedSections {
     }
     $sections{HDAT}{out}       = "$scratch_dir/hdat.bin.ecc";
 
+    # Populate OCMBFW partition if it exists in the layout
+    if(checkForPnorPartition("OCMBFW", $parsed_pnor_layout))
+    {
+        if(-e $ocmbfw_original_filename)
+        {
+            #Verify image validity
+            run_command("$hb_image_dir/pkgOcmbFw.pl --verify --packagedBin $ocmbfw_original_filename");
+            $sections{OCMBFW}{in}    = "$ocmbfw_original_filename";
+        }
+        else
+        {
+            print "WARNING: OCMBFW binary not found, generating blank binary (w/ valid header) instead\n";
+            #Create blanke 4k image
+            my $ocmbfw_generated_filename = "ocmbfw_generated.bin";
+            run_command("dd if=/dev/zero of=$scratch_dir/$ocmbfw_generated_filename bs=1024 count=4");
+        
+            #Add header to blank image
+            my $date = `date`;
+            run_command("$hb_image_dir/pkgOcmbFw.pl --unpackagedBin $scratch_dir/$ocmbfw_generated_filename --packagedBin $ocmbfw_original_filename --timestamp \"$date\" --vendorVersion \"0.1\" --vendorUrl \"http://www.ibm.com\"");
+
+            # verify header sha512 hash value matches value calculated against image
+            run_command("$hb_image_dir/pkgOcmbFw.pl --verify --packagedBin $ocmbfw_original_filename");
+            $sections{OCMBFW}{in}    = "$ocmbfw_original_filename";
+        }
+        #Final image will be under a new name after ECC protection and any other processing required
+        $sections{OCMBFW}{out}       = "$ocmbfw_binary_filename";
```

Comment:
why wouldn't this just be hardcoded to "$scratch_dir/obmcfw.bin.ecc" like the other partitions?  it seems to be a standard to just have the name me <partition>_bin.ecc.

Also, are you going to miss the $scratch_dir part of the name?

Reply:
Because the other places have bugs. If a user actually defines some of those config variables to something other than the default this area would fail. My method would not.

Reply:
are you still missing the $scratch_dir part of the name, though?

Reply:
Bill pointed me to the op-build change where the $scratch_dir is already part of the file name:
https://github.com/open-power/op-build/pull/2929/files